services/gemma-local/src/services/model_downloader.py

```
@@ -0,0 +1,125 @@
+
+            logger.info(f"Configuration updated to use {model_id}")
+
+        except Exception as e:
+            logger.error(f"Model download failed: {e}", exc_info=True)
+            yield DownloadProgress(
+                status=ModelStatus.ERROR,
+                message="Download failed",
+                progress=0.0,
+                error="Download failed",
+            )
+            raise ModelDownloadError(model_name, "Download failed")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)